blockchain/fullblocktests/generate.go

```diff
@@ -543,6 +544,17 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	tests = append(tests, testInstances)
 	g.AssertTipHeight(uint32(coinbaseMaturity) + 1)
 
+	// TODO: Test is not failing as expected
```

Comment:
This is failing for several reasons.

1) You can't spend anything this early since nothing is mature yet.
2) You're not adding the transaction created to the stake transaction tree
3) You're not incrementing `FreshStake` in the header to commit to the purchase